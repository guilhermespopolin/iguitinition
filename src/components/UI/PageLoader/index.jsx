import React from 'react'
import PropTypes from 'prop-types'

import styles from './styles.css'

function PageLoader({ show }) {
  if (!show) {
    return null
  }

  return (
    <div className={styles.wrapper}>
      <div className={styles.rubikLoader} />
    </div>
  )
}

PageLoader.propTypes = {
  show: PropTypes.bool,
}

PageLoader.defaultProps = {
  show: true,
}

export default PageLoader
