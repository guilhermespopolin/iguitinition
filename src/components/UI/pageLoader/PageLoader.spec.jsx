import React from 'react';
import renderer from 'react-test-renderer';

import PageLoader from '.';

describe('<PageLoader />', () => {
  describe('When error prop iquals true', () => {
    const shallowRendered = renderer.create(
      <PageLoader error={new Error('error')} />,
    ).toJSON();

    it('should renders properly', () => {
      expect(shallowRendered).toMatchSnapshot();
    });
  });

  describe('When pastDelay prop iquals to true', () => {
    const shallowRendered = renderer.create(<PageLoader pastDelay />).toJSON();

    it('should renders properly', () => {
      expect(shallowRendered).toMatchSnapshot();
    });
  });

  describe('When there is no error and pastDelay prop iquals to false', () => {
    const shallowRendered = renderer.create(<PageLoader />).toJSON();

    it('should renders properly', () => {
      expect(shallowRendered).toMatchSnapshot();
    });
  });
});
