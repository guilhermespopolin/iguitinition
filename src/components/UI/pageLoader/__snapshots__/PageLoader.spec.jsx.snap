// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLoader /> When error prop iquals true should renders properly 1`] = `
<h1>
  Something went wrong!!
</h1>
`;

exports[`<PageLoader /> When pastDelay prop iquals to true should renders properly 1`] = `
<div
  className="wrapper"
>
  <div
    className="rubikLoader"
  />
</div>
`;

exports[`<PageLoader /> When there is no error and pastDelay prop iquals to false should renders properly 1`] = `null`;
